src/components/TodoContainer.js

```diff
@@ -0,0 +1,101 @@
+import React, { useState, useEffect, useCallback } from 'react';
+import TodoList from './TodoList';
+import AddTodoForm from './AddTodoForm';
+import TodoListToggle from './TodoListToggle';
+import api from './Api';
+import style from './App.css';
+import PropTypes from 'prop-types';
+
+const TodoContainer = ({ tableName }) => {
+  const [todoList, setTodoList] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [checkedState, setCheckedState] = useState({});
+  const [isAscending, setIsAscending] = useState(true);
+
+  const fetchAndSortTodos = useCallback(async () => {
+    const sortField = 'title'; 
+    const sortOptions = `sort[0][field]=${sortField}&sort[0][direction]=asc`;
+
+    const sortedTodos = await api.fetchAndSortTodos(tableName, sortOptions);
+
+    setTodoList(sortedTodos);
+    setIsLoading(false);
+  }, [tableName]);
+
+  useEffect(() => {
+    fetchAndSortTodos();
+  }, [fetchAndSortTodos]);
+
+  const toggleSortOrder = () => {
+    setIsAscending(!isAscending);
+
+    const sortedTodos = [...todoList].sort((a, b) => {
+      if (isAscending) {
+        return a.title.localeCompare(b.title);
+      } else {
+        return b.title.localeCompare(a.title);
+      }
+    });
+
+    setTodoList(sortedTodos);
+  };
+
+  const removeTodo = async (id) => {
+    const response = await api.removeTodo(tableName, id);
+    if (response === id) {
+      setTodoList((prevTodoList) => prevTodoList.filter((todo) => todo.id !== id));
+    }
+  };
+
+  const addTodo = async (title) => {
+    const response = await api.addTodo(tableName, title);
+    if (response !== undefined) {
```

Comment:
because of "Truthy "and "Falsy" in javascript, you can often use a single value inside of an if, because every value will either be considered equivalent to `true` or `false`: 

I don't know if it's really a good thing tbh, but you'll usually see the convention for things like this as:

```suggestion
    // response is not `null`, `undefined`, `false` or any other "falsy" value
    if (response) {
```
https://developer.mozilla.org/en-US/docs/Glossary/Truthy
https://developer.mozilla.org/en-US/docs/Glossary/Falsy

Reply:
actually now that I think about it, it feels like this check should be able to happen inside of `api.addTodo()`, so the file which calls your function (aka its "consumer") can do less of those tedious error checks